server/src/remote.rs

```diff
@@ -545,6 +548,55 @@ impl FlightService for FlightServiceHandler {
 
             // Confirm the data was flushed.
             Ok(Response::new(Box::pin(stream::empty())))
+        } else if action.r#type == "CollectMetrics" {
+            let mut storage_engine = self.context.storage_engine.write().await;
+            let metrics = storage_engine.collect_metrics().await;
+
+            // Extract the data from the metrics and insert it into Apache Arrow array builders.
+            let mut metric_builder = StringBuilder::new();
+            let mut timestamps_builder = ListBuilder::new(TimestampBuilder::new());
+            let mut values_builder = ListBuilder::new(UInt32Builder::new());
+
+            for (metric_name, (timestamps, values)) in metrics.iter() {
+                metric_builder.append_value(metric_name);
+
+                timestamps_builder
+                    .values()
+                    .append_slice(timestamps.values());
+                timestamps_builder.append(true);
+
+                values_builder.values().append_slice(values.values());
+                values_builder.append(true);
+            }
+
+            let schema = self.context.metadata_manager.metric_schema();
+
+            // Finish the builders and create the record batch containing the metrics.
+            let batch = RecordBatch::try_new(
+                schema.0.clone(),
```

Comment:
Unclear why this `.clone()` is needed when `metric_schema()` returns `self.metric_schema.clone()`.

Reply:
Removing this `clone()` causes an error since the same schema is used in `StreamWriter::try_new_with_options`. One way to solve this would be to call `self.context.metadata_manager.metric_schema()` twice but I think that would have the exact same effect as calling it once and cloning the result.